### ISSUE TITLE
Bayonetta 2

### ISSUE THREAD

**Issue by dima-xd** [strato-emu/compat-list#582]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Tested Build

Edge 32

### Android Version

12

### SoC type

Snapdragon

### Device GSMArena link

https://www.gsmarena.com/xiaomi_poco_f3-10758.php

### Game Behaviour

FINISHED BY ROBMAGICKER

- Minor bugs, like shading off during cutscenes.

![Screenshot_Skyline_Edge_2023-02-09-19-01-10-163](https://user-images.githubusercontent.com/81425424/218287083-4d5ba973-37d9-4898-ae56-882c09d24a55.jpg)

![Screenshot_Skyline_Edge_2023-02-09-18-26-28-072](https://user-images.githubusercontent.com/81425424/218287051-d611fc00-f8b5-441d-8441-8b1a067917cb.jpg)
.jpg)

### Is Log Level is set to 'Info'?

- [X] Yes

### Log file

[emulation.zip](https://github.com/skyline-emu/skyline-games-list/files/8872123/emulation.zip)

**Comment by Crytonics** [comment]
```
E1281HOS-0array::at (Service: IHidServer::SetSupportedNpadIdType)
Stack Trace:
* 0x7C51408F2C (skyline::service::BaseService::HandleRequest(skyline::kernel::type::KSession&, skyline::kernel::ipc::IpcRequest&, skyline::kernel::ipc::IpcResponse&) from libskyline.so)
* 0x7C51407C8C (skyline::service::ServiceManager::SyncRequestHandler(unsigned int) from libskyline.so)
* 0x7C5134ECA4 (skyline::kernel::svc::SendSyncRequest(skyline::DeviceState const&) from libskyline.so)
* 0x7C5132CC20 (skyline::nce::NCE::SvcHandler(unsigned short, skyline::nce::ThreadContext*) from libskyline.so)
* 0x8055ED0B8 (from sdk.nso.patch)
* 0x80580A774 (nn::sf::hipc::client::Hipc2ClientCoreProcessor<(nn::sf::hipc::detail::MessageType)4, nn::sf::cmif::detail::CoreMethodInfo<std::__1::tuple<nn::sf::cmif::detail::InRawArgumentInfo<8ul, 8ul, 0l>, nn::sf::cmif::detail::BufferArgumentInfo<0, 9, 0ul> >, 8ul, 0ul, true>, std::__1::tuple<nn::sf::cmif::detail::InRawArgumentInfo<8ul, 8ul, 0l>, nn::sf::cmif::detail::BufferArgumentInfo<0, 9, 0ul> > >::Process(nn::sf::hipc::client::Hipc2ClientProxyBaseObject*, nn::sf::cmif::client::InRawClientArgumentType<8ul, 8ul, void>, nn::sf::cmif::client::BufferClientArgumentType, unsigned int) from sdk.nso)
* 0x8058096FC (nn::sf::cmif::client::detail::CmifProxyImpl<nn::hid::IHidServer, nn::sf::hipc::client::Hipc2ProxyKindBase<(nn::sf::hipc::detail::MessageType)4>, nn::sf::StatelessAllocationPolicy<nn::sf::ExpHeapStaticAllocator<16384ul, nn::hid::detail::HidServer> >, nn::hid::IHidServer>::_nn_sf_sync_SetSupportedNpadIdType(nn::applet::AppletResourceUserId, nn::sf::InArray<unsigned int> const&) from sdk.nso)
* 0x8057F1F48 (nn::hid::detail::SetSupportedNpadIdType(unsigned int const*, int) from sdk.nso)
* 0x805802BA0 (nn::hid::SetSupportedNpadIdType(unsigned int const*, unsigned long) from sdk.nso)
* 0x800DA42B4 (from main.nso)
* 0x800DA5510 (from main.nso)
* 0x80096F9F4 (from main.nso)
* 0x800997168 (from main.nso)
* 0x800996E90 (nnMain from main.nso)
* 0x8056576B4 (nn::init::Start(unsigned long, unsigned long, void (*)(), void (*)()) from sdk.nso)
* 0x8000010AC (from rtld.nso)
```

**Comment by dimaxdqwerty** [comment]
So, the problem is in `void NpadManager::Update()` method. For some reason element from `std::vector<NpadId> supportedIds` has weird value (see screenshot) and when it casts it to `static_cast<size_t>(id)` it dies on `npads.at(Translate(id))` because array is not too long ofc.
![image](https://user-images.githubusercontent.com/70838468/198822520-881c918f-d6a0-40df-ba79-8fc69758d3c0.png)
If devs will fix it, then it will boot in game normally on gpu-new (with some graphics problem and ~20 fps)

**Comment by dimaxdqwerty** [comment]
@bylaws please look at my comment. Sorry for pinging, just want to let you know and I hope you will fix it

**Comment by guranu** [comment]
@bylaws please change the status from status-nothing to status-ingame the is now booting with very decent framerate.
![Screenshot_20221101-141029](https://user-images.githubusercontent.com/103632948/199220023-dde04297-3a51-4e95-8340-986ee2a7ec8e.jpg)
![Screenshot_20221101-141040](https://user-images.githubusercontent.com/103632948/199220031-f58e4d06-5ffe-4d41-a02d-7c82f8626eff.jpg)
![Screenshot_20221101-141008](https://user-images.githubusercontent.com/103632948/199220033-77241005-8d50-41b9-9563-db274e0e3e35.jpg)

**Comment by guranu** [comment]
@Crytonics please change the status from status-nothing to status-ingame.

**Comment by Ravenqueen60** [comment]
how can i download it?

**Comment by kashif102** [comment]
It's working with latest v24 turnip drivers 15 to 25 fps 🥲
![Screenshot_2022-11-23-18-13-32-525_com miui videoplayer](https://user-images.githubusercontent.com/118906198/203555588-25921f65-8299-4d87-8302-4d04fafd227b.jpg)

**Comment by hstr0100** [comment]
I don't think it should be considered "playable" if it does not run with the default drivers.
This game does not boot on Qualcomm v0.408.0/Snapdragon 845, stops at the blue splashscreen before the game title appears.

I am completely against the current trend of having to join discord servers to have access to basic functionality, the act of having to download a separate file from a different source so that certain games will work.

It could be that my specific device is having issues with this game in particular, but overall I just think that if it does not run out of the box with the latest build of the emulator, then in my opinion it should be labelled differently.
I hope this opinion can be taken in consideration if valid.

**Comment by Swizzlefowl** [comment]
Your default driver might be too old to render the game correctly, and it's fine to label it playable if it's playable with a different driver as you would for example just update your drivers if the game doesn't function correctly if you were using a PC. Unfortunately it's not as easy to update drivers in Android devices but it's possible to use different drivers through adrenotools. Also you don't have to join our discord server if you don't want to you can get drivers from our driver repository https://github.com/K11MCH1/AdrenoToolsDrivers.

**Comment by Ahmed123Jamali** [comment]
Not working tekno sparks 8 pro

**Comment by lnfel** [comment]
> Your default driver might be too old to render the game correctly, and it's fine to label it playable if it's playable with a different driver as you would for example just update your drivers if the game doesn't function correctly if you were using a PC. Unfortunately it's not as easy to update drivers in Android devices but it's possible to use different drivers through adrenotools. Also you don't have to join our discord server if you don't want to you can get drivers from our driver repository https://github.com/K11MCH1/AdrenoToolsDrivers.

There should be a documentation of this on the main website. That is a huge bit of info people would be missing out especially if it was their first time emulating on android. Egg NS had this sort of thing available as a menu setting which download the drivers.

**Comment by Arka161** [comment]
Egg promotes piracy, and should not be used as an example
